src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/segmentgroup/SegmentGroup.java

```diff
@@ -0,0 +1,39 @@
+package uk.nhs.digital.nhsconnect.lab.results.model.edifact.segmentgroup;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.Section;
+
+public abstract class SegmentGroup extends Section {
+    public SegmentGroup(final List<String> edifactSegments) {
+        super(edifactSegments);
+    }
+
+    public abstract String getKey();
+
+    protected static List<List<String>> splitMultipleSegmentGroups(final List<String> edifactSegments,
+                                                                   final String key) {
+        if (!edifactSegments.get(0).startsWith(key)) {
+            return Collections.emptyList();
+        }
```

Comment:
If no segment group headers were found by the calling code, return nothing. This might also need to check for empty.